lib/MongoDB/Collection.pm

```diff
@@ -1064,6 +1064,8 @@ sub aggregate {
     my ( $self, $pipeline, $options ) = @_;
     $options ||= {};
 
+    my $session = delete $options->{session};
```

Comment:
Elsewhere, I suggest that we do all the session validation here at the borders of the API.  I was thinking that we do something along the lines of:

```
my $session = $self->_client->maybe_use_session(delete $options->{session});
```

Where `maybe_use_session` either validates that an explicit session comes from that client and returns it, or returns an implicit session or returns undef if sessions aren't supported by the topology.  Then below in constructing the op, we can unconditionally pass in `session -> $session` and it's either a valid session or undef if not supported.  Later, ops can do the right thing either way.

I think doing that defensive validation/implicit session acquisition at the border of the app is going to simplify the code substantially with only a little bit of boilerplate in all the public API methods.  Everywhere else can assume that if the 'session' in an op is undef, then sessions (and cluster time) are unsupported.  If there is a session object, then ops can also reach the client object (e.g. for updating the cluster time).